src/lib/serverPlugins/entities.js

```diff
@@ -0,0 +1,280 @@
+      if (entity.type == 'player') {
+        return {
+          entityId: entity.id,
+          playerUUID: entity.player._client.uuid,
+          x: entity.position.x,
+          y: entity.position.y,
+          z: entity.position.z,
+          yaw: entity.yaw,
+          pitch: entity.pitch,
+          currentItem: 0,
+          metadata: entity.metadata
+        }
+      } else if (entity.type == 'object') {
+        return {
+          entityId: entity.id,
+          type: entity.entityType,
+          x: entity.position.x,
+          y: entity.position.y,
+          z: entity.position.z,
+          pitch: entity.pitch,
+          yaw: entity.yaw,
+          objectData: {
+            intField: entity.data,
+            velocityX: scaledVelocity.x,
+            velocityY: scaledVelocity.y,
+            velocityZ: scaledVelocity.z
+          }
+        }
+      } else if (entity.type == 'mob') {
+        return {
+          entityId: entity.id,
+          type: entity.entityType,
+          x: entity.position.x,
+          y: entity.position.y,
+          z: entity.position.z,
+          yaw: entity.yaw,
+          pitch: entity.pitch,
+          headPitch: entity.headPitch,
+          velocityX: scaledVelocity.x,
+          velocityY: scaledVelocity.y,
+          velocityZ: scaledVelocity.z,
+          metadata: entity.metadata
+        }
+      }
+    };
+
+    entity.getNearby = () => serv
+      .getNearbyEntities({
+        world: entity.world,
+        position: entity.position,
+        radius: entity.viewDistance*32
+      })
+      .filter((e) => e != entity);
+
+    entity.updateAndSpawn = () => {
+      var updatedEntities=entity.getNearby();
+      var entitiesToAdd=updatedEntities.filter(e => entity.nearbyEntities.indexOf(e)==-1);
+      var entitiesToRemove=entity.nearbyEntities.filter(e => updatedEntities.indexOf(e)==-1);
+      if (entity.type == 'player') {
+        entity.player.despawnEntities(entitiesToRemove);
+        entitiesToAdd.forEach(entity.player.spawnEntity);
+        entity.player.lastPositionPlayersUpdated=entity.position.clone();
+      } else {
+        entity.lastPositionPlayersUpdated=entity.position.clone();
+      }
+
+      var playersToAdd = entitiesToAdd.filter(e => e.type == 'player').map(e => e.player);
+      var playersToRemove = entitiesToRemove.filter(e => e.type == 'player').map(e => e.player);
+
+      playersToRemove.forEach(p => p.despawnEntities([entity]));
+      playersToRemove.forEach(p => p.entity.nearbyEntities=p.entity.getNearby());
+      playersToAdd.forEach(p => p.spawnEntity(entity));
+      playersToAdd.forEach(p => p.entity.nearbyEntities=p.entity.getNearby());
+
+      entity.nearbyEntities=updatedEntities;
+    };
+
+    return entity;
+  }
+
+  serv.spawnObject = (type, world, position, {pitch=0,yaw=0,velocity=vec3(0,0,0),data=1,itemId,itemDamage=0}={}) => {
```

Comment:
better call itemId,itemDamage as type and metadata as we do in the rest of flying-squid